daemon/src/file_asset_source.rs

```diff
@@ -394,233 +446,278 @@ impl FileAssetSource {
         }
         Ok(())
     }
-    fn check_for_importer_changes(&self) -> Result<bool> {
-        let mut changed_paths = Vec::new();
-        {
-            let txn = self.db.ro_txn()?;
-            for result in self.iter_metadata(&txn)? {
-                let (path, metadata) = result?;
-                let metadata = metadata.get()?;
-                let changed = if let Some(importer) = self.importers.get_by_path(&path) {
-                    metadata.get_importer_version() != importer.version()
-                        || metadata.get_importer_options_type()?
-                            != importer.default_options().uuid()
-                        || metadata.get_importer_state_type()? != importer.default_state().uuid()
-                        || metadata.get_importer_type()? != importer.uuid()
-                } else {
-                    false
-                };
+
+    fn check_for_importer_changes(&self) -> bool {
+        let txn = self.db.ro_txn().expect("Failed to open ro txn");
+
+        let changed_paths: Vec<PathBuf> = self
+            .iter_metadata(&txn)
+            .filter_map(|(path, metadata)| {
+                let metadata = metadata.get().ok()?;
+                let changed = self
+                    .importers
+                    .get_by_path(&path)
+                    .and_then(|importer| {
+                        let importer_version = metadata.get_importer_version();
+                        let options_type = metadata.get_importer_options_type().ok()?;
+                        let state_type = metadata.get_importer_state_type().ok()?;
+                        let importer_type = metadata.get_importer_type().ok()?;
+
+                        Some(
+                            importer_version != importer.version()
+                                || options_type != importer.default_options().uuid()
+                                || state_type != importer.default_state().uuid()
+                                || importer_type != importer.uuid(),
+                        )
+                    })
+                    .unwrap_or(false);
```

Comment:
These errors seem to be related to reading the capnp messages, and I think we should panic if these types of errors ever happen, because they shouldn't. Probably indicates a corrupt database or has old data that has not been upgraded to newer versions.